src/components/contact-form/contactForm.js

```diff
@@ -0,0 +1,167 @@
+import { useState } from 'react';
+import styles from './contactForm.module.scss';
+
+const config = {
+  script:
+    'https://script.google.com/macros/s/AKfycbx9XF_VGYdZQoALe4UH3TFyyVMsbENMPMVwp7ZORg/exec',
+  sheet: 'Contact Form Responses',
+  email: 'chanpark93@gmail.com',
+};
+
+const Thanks = ({ name }) => (
+  <div className={styles.thankyou}>
+    <h2>Thanks for contacting us, {name}! We will get back to you soon!</h2>
+  </div>
+);
+
+export default () => {
+  const initialForm = {
+    name: '',
+    organization: '',
+    email: '',
+    phone: '',
+    subject: '',
+    message: '',
+    honeypot: '',
+  };
+
+  const [info, setInfo] = useState(initialForm);
+
+  const { name, organization, email, phone, subject, message, honeypot } = info;
+
+  const initialState = {
+    sent: false,
+    clicked: false,
+  };
+
+  const [state, setState] = useState(initialState);
+
+  const { sent, clicked } = state;
+
+  const handleChange = (e) => {
+    e.preventDefault();
+    setInfo({ ...info, [e.target.name]: e.target.value });
+  };
+
+  const handleClick = (e) => {
+    e.preventDefault();
+    // loading
+    setState({ clicked: true });
+
+    //sent
+    const setSent = () => {
+      setState({ sent: true });
+    };
+
+    // check if bot
+    if (honeypot) return false;
+
+    // sending through xhr as axios won't work (CORS)
+    const xhr = new XMLHttpRequest();
+    xhr.open('POST', config.script);
+    // xhr.withCredentials = true
+    xhr.setRequestHeader('Content-Type', 'application/x-www-form-urlencoded');
+    xhr.onreadystatechange = function xhrGo() {
+      setSent();
+    };
+
+    const encoded = Object.keys(info)
+      .map((k) => {
+        return `${encodeURIComponent(k)}=${encodeURIComponent(info[k])}`;
+      })
+      .join('&');
+
+    // finally sending
+    xhr.send(encoded);
```

Comment:
NextJS has a way of handling CORS issues:

https://nextjs.org/docs/api-routes/api-middlewares#connectexpress-middleware-support

There's also a simpler way to send form data using **node-fetch** package:

https://github.com/node-fetch/node-fetch#post-with-form-parameters

Reply:
This was actually a copy paste from 

https://github.com/dwyl/learn-to-send-email-via-google-script-html-no-server#how

--

I did this to save time since Ryan wants this up and running as soon as possible, but I'll come back to updating it with your suggestion later down the road, since that will take me some time to learn haha